src/main/java/io/supertokens/webserver/api/session/JWTDataAPI.java

```diff
@@ -55,6 +59,11 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IO
         JsonObject userDataInJWT = InputParser.parseJsonObjectOrThrowError(input, "userDataInJWT", false);
         assert userDataInJWT != null;
 
+        if (getVersionFromRequest(req).greaterThanOrEqualTo(SemVer.v2_21) &&
+                Arrays.stream(protectedPropNames).anyMatch(userDataInJWT::has)) {
+            throw new ServletException(new BadRequestException("The user payload contains protected field"));
+        }
```

Comment:
these checks should not happen here, instead, update updateSession to have this check, and create another updateSessionBeforeCDI2_21 without the check and mark that as deprecated